### PR TITLE
Rename Container to BlobContainer

### LINKED ISSUE #7835: Storage: Make sure `Container` is always written `BlobContainer`
Our client is `BlobContainerClient` but a number of APIs (like `ContainerSasPermissions`) don't have `Blob` as a prefix.  We should be consistent and make it `BlobContainer` everywhere.

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Fixes #7835

### CODE REVIEW

File: sdk/storage/Azure.Storage.Blobs/src/BlobServiceClient.cs [hunk 1]
```diff
@@ -344,7 +344,7 @@ internal BlobServiceClient(Uri serviceUri, HttpPipeline pipeline)
                     $"{nameof(options)}: {options}");
                 try
                 {
-                    return await BlobRestClient.Service.ListContainersSegmentAsync(
+                    return await BlobRestClient.Service.ListBlobContainersSegmentAsync(
```
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Blobs/src/BlobServiceClient.cs]: This is internal and doesn't really need to be renamed

File: sdk/storage/Azure.Storage.Blobs/src/Models/GetBlobsOptions.cs [hunk 2]
```diff
@@ -15,12 +15,15 @@ namespace Azure.Storage.Blobs.Models
     /// </summary>
     public struct GetBlobsOptions : IEquatable<GetBlobsOptions>
     {
+        // we can get rid of whole type by making prefix and enums parameters to wherever this type is currently used
+
```
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Blobs/src/Models/GetBlobsOptions.cs]: Should revert

**Review by tg-msft — APPROVED** [review]
Can you do a quick Find All in Visual Studio for `(?<!Blob)Container` using case sensitivity and regexes in your *.cs files?

**Review by tg-msft — APPROVED** [review]
Looks good.

FYI @seanmcc-msft and @amnguye - I'd like to merge this today so it'll cause as few conflicts as possible.  Mind taking a quick skim?  It's a very mechanical global rename per API Review feedback.

**Review by amnguye — APPROVED** [review]

**Pull request merged**